src/components/v-mapbox-layer.js

```diff
@@ -31,16 +31,9 @@ export default {
     }
   },
 
-  data: () => ({
-    isInitialized: false
-  }),
-
   methods: {
     deferredMountedTo() {
-      if(!this.isInitialized) {
-        this.renderLayer();
-        this.isInitialized = true;
-      }
+      this.renderLayer();
```

Comment:
If the map style changed `isInitialized` was true, so the layers weren't being re-rendered.